encoding/codecv2/codecv2.go

```diff
@@ -222,6 +235,46 @@ func ConstructBlobPayload(chunks []*encoding.Chunk, useMockTxData bool) (*kzg484
 	return blob, blobVersionedHash, &z, nil
 }
 
+// DecodeTxsFromBlob decodes txs from blob bytes and writes to chunks
+func DecodeTxsFromBlob(blob *kzg4844.Blob, chunks []*DAChunkRawTx) error {
+	compressedBytes := codecv1.BytesFromBlobCanonical(blob)
+	magics := []byte{0x28, 0xb5, 0x2f, 0xfd}
+
+	blobBytes, err := decompressScrollBatchBytes(append(magics, compressedBytes[:]...))
+	if err != nil {
+		return err
+	}
+
+	numChunks := int(binary.BigEndian.Uint16(blobBytes[0:2]))
```

Comment:
I think it will help introducing a new function: `DecodeTxsFromBytes`, where its implementation does not change between v1 and v2.

The `DecodeTxsFromBlob` can internally call its own version's `DecodeTxsFromBytes` after translating the blob to bytes.

In v1 that would be: canonical blob -> blob bytes
In v2 that would be: canonical blob -> blob bytes -> decompress

Reply:
As a result we can then remove the code duplication between v1 and v2 in the `DecodeTxsFromBlob` function. In the future if we detect a bug, it would help updating/reviewing as there would be lesser number of code changes.